src/internal/primitives/FocusTrap.tsx

```diff
@@ -0,0 +1,83 @@
+import type React from 'react';
+import { useEffect, useRef } from 'react';
+
+const FOCUSABLE_ELEMENTS_SELECTOR =
+  'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
```

Comment:
You probably want to add `:not([disabled])` pseudo classes to the button, input, select, and textarea elements. Like:

`button:not([disabled])`
```suggestion
  'button:not([disabled]), [href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"])';
```

Reply:
Good call